domain/access/modelmigration/import.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package modelmigration
+
+import (
+	"context"
+	"time"
+
+	"github.com/juju/description/v8"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/logger"
+	coremodel "github.com/juju/juju/core/model"
+	"github.com/juju/juju/core/modelmigration"
+	corepermission "github.com/juju/juju/core/permission"
+	"github.com/juju/juju/core/user"
+	accesserrors "github.com/juju/juju/domain/access/errors"
+	"github.com/juju/juju/domain/access/service"
+	"github.com/juju/juju/domain/access/state"
+)
+
+// Coordinator is the interface that is used to add operations to a migration.
+type Coordinator interface {
+	// Add adds the given operation to the migration.
+	Add(modelmigration.Operation)
+}
+
+// RegisterImport registers the import operations with the given coordinator.
+func RegisterImport(coordinator Coordinator, logger logger.Logger) {
+	coordinator.Add(&importOperation{
+		logger: logger,
+	})
+}
+
+// ImportService provides a subset of the access domain
+// service methods needed for model permissions import.
+type ImportService interface {
```

Comment:
**todo:** use [godoc doclink](https://tip.golang.org/doc/comment#doclinks) for errors